packages/engine/Source/Scene/SpecularEnvironmentCubeMap.js

```diff
@@ -0,0 +1,233 @@
+import defined from "../Core/defined.js";
+import destroyObject from "../Core/destroyObject.js";
+import Event from "../Core/Event.js";
+import loadKTX2 from "../Core/loadKTX2.js";
+import PixelFormat from "../Core/PixelFormat.js";
+import CubeMap from "../Renderer/CubeMap.js";
+import PixelDatatype from "../Renderer/PixelDatatype.js";
+import Sampler from "../Renderer/Sampler.js";
+import TextureMinificationFilter from "../Renderer/TextureMinificationFilter.js";
+
+/**
+ * TODO
+ *
+ * @private
+ */
+function SpecularEnvironmentCubeMap(url) {
```

Comment:
It would be great to decouple this from KTX2 loading as much as possible. I think we can maintain that functionality in this class still, just under a `SpecularEnvironmentCubeMap.fromKTX2` factory function instead.

That function would be asynchronous. It would load the KTX2 buffers, then create and return a `SpecularEnvironmentCubeMap`, ie.

```js
SpecularEnvironmentCubeMap.fromKTX2 = async function(url) {
    const buffers = await loadKTX2(url);
    ...
    return new SpecularEnvironmentCubeMap(buffers, ...);
}
```

Reply:
As discussed: this change would make a lot of sense, and could simplify the IBL flow overall. However, it is a significant refactor and probably out of scope for this PR.
I opened https://github.com/CesiumGS/cesium/issues/12113 to discuss further.